### PR TITLE
Homework2: Eldar Dniaprouski

### PR DESCRIPTION
**Pull request by eldar-dniaprouski**

### CODE REVIEW

**Review by belovmd — COMMENTED** [review]
1. please remove tasks from first homework

File: src/Homework_02/02_Examples(py.checkio.org).py [hunk 1]
```diff
@@ -0,0 +1,120 @@
+"""
+1. First task
```
Comment (belovmd) [inline src/Homework_02/02_Examples(py.checkio.org).py:2]: please put tasks description

File: src/Homework_02/02_Examples(py.checkio.org).py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+5.You are given a positive integer.
+Your function should calculate the product of the digits excluding any zeroes.
+"""
+
+
+def checkio(number: int) -> int:
+    result = 1
+    for i in str(number):
+        n = int(i)
+        if n != 0:
```
Comment (belovmd) [inline src/Homework_02/02_Examples(py.checkio.org).py]: if not n

File: src/Homework_02/02_Examples(www.codewars.com).py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+Implement a function that determines
+whether a string that contains only letters is an isogram.
+Assume the empty string is an isogram. Ignore letter case.
+"""
+
+
+def is_isogram(string):
+    my_low_string = string.lower()
+    my_new_string = ''
+    for i in range(len(my_low_string)):
```
Comment (belovmd) [inline src/Homework_02/02_Examples(www.codewars.com).py]: for element in my_low_string:

File: src/Homework_02/02_Examples(www.hackerrank.com).py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+The third line contains the product of the two numbers.
+1 <= a <= 10 ** 10
+1 <= b <= 10 ** 10
+"""
+
+
+if __name__ == '__main__':
+    a = int(input())
+    b = int(input())
+    if (1 <= a <= 10 ** 10) and (1 <= b <= 10 ** 10):
```
Comment (belovmd) [inline src/Homework_02/02_Examples(www.hackerrank.com).py]: redundant checks - it's put to string description

File: src/Homework_02/02_Examples(www.hackerrank.com).py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+"""
+3. You are given a string and your task is to swap cases.
+In other words, convert all lowercase letters to uppercase letters
+and vice versa.
+"""
+
+
+def swap_case(s):
+    s_new = ''
+    if 1 < len(s) <= 1000:
```
Comment (belovmd) [inline src/Homework_02/02_Examples(www.hackerrank.com).py]: why do we need this check?

File: src/Homework_02/02_Examples(www.hackerrank.com).py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+        for i in range(len(s)):
+            if s[i] == s[i].lower():
```
Comment (belovmd) [inline src/Homework_02/02_Examples(www.hackerrank.com).py]: islower isupper methods for example

**Comment by eldar-dniaprouski** [comment]
Removed tasks from first homework

File: src/Homework_02/02_Examples(www.hackerrank.com).py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+You have to print the number of times
+that the substring occurs in the given string.
+String traversal will take place from left to right,
+not from right to left.
+"""
+
+
+def count_substring(string, sub_string):
+    num_count = 0
+    if 1 <= len(string) <= 200:
```
Comment (belovmd) [inline src/Homework_02/02_Examples(www.hackerrank.com).py]: redundant

File: src/Homework_02/02_Examples(www.hackerrank.com).py [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+        for i in range(len(string)):
```
Comment (belovmd) [inline src/Homework_02/02_Examples(www.hackerrank.com).py]: you can use counter, while cycle and .find() method

File: src/Homework_02/02_Examples(www.hackerrank.com).py [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+5. You are asked to ensure that the first and last names of people
+begin with a capital letter in their passports.
+For example, alison heck should be capitalised correctly as Alison Heck.
+"""
+
+
+def solve(s):
+    s_new = ''
+    s_new += s[0].upper()
+    if 0 < len(s) < 1000:
```
Comment (belovmd) [inline src/Homework_02/02_Examples(www.hackerrank.com).py]: redundant

File: src/Homework_02/02_Examples(www.hackerrank.com).py [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+        for i in range(1, len(s)):
```
Comment (belovmd) [inline src/Homework_02/02_Examples(www.hackerrank.com).py]: split, for loop and capitalize method ^:)

File: src/Homework_02/03_Max_Dlina.py [hunk 11]
```diff
@@ -0,0 +1,28 @@
+"""
+3. Найти самое длинное слово в введенном предложении.
+Учтите что в предложении есть знаки препинания.
+"""
+
+my_string = 'aaa bbbb! aa, bbb! c bbbaa bb b. b? aaa'
+print(my_string, '\n')
+
+# не самый красивый вариант
+new_string = my_string.replace(
```
Comment (belovmd) [inline src/Homework_02/03_Max_Dlina.py]: what about ?(){} etc? 
you can use string with punctuation characters and using for loop replace all of them

File: src/Homework_02/03_Max_Dlina.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+
+# вариант лучше
+table = str.maketrans("", "", "!?,.")
+new_string2 = my_string.translate(table)
+print(new_string2, '\n')
+
+my_split_string = new_string2.split()
+max_len = 0
+max_len_index = 0
+for i in range(len(my_split_string)):
```
Comment (belovmd) [inline src/Homework_02/03_Max_Dlina.py]: for word in my_split:

File: src/Homework_02/03_Max_Dlina.py [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+    cur_len = len(my_split_string[i])
+    print(my_split_string[i], 'длина -', cur_len)
```
Comment (belovmd) [inline src/Homework_02/03_Max_Dlina.py]: we don't need to print all words but longest

File: src/Homework_02/04_Povtor.py [hunk 14]
```diff
@@ -0,0 +1,15 @@
...
+Требуется удалить из нее повторяющиеся символы и все пробелы.
+Например, если было введено "abc cde def", то должно быть выведено "abcdef".
+"""
+
+# my_string = input('Введите строку символов через пробел\n')
+my_string = 'abc cdde edeff'
+my_rep_string = my_string.replace(' ', '')
+print(my_rep_string)
+my_new_string = ''
+for i in range(len(my_rep_string)):
```
Comment (belovmd) [inline src/Homework_02/04_Povtor.py]: range(len(iterable)) usually is  a bad idea

File: src/Homework_02/04_Povtor.py [hunk 15]
```diff
@@ -0,0 +1,15 @@
...
+    if my_new_string.find(my_rep_string[i]) == -1:
```
Comment (belovmd) [inline src/Homework_02/04_Povtor.py]: if character in my_new_string:

File: src/Homework_02/05_Bukvy.py [hunk 16]
```diff
@@ -0,0 +1,19 @@
+"""
+5. Посчитать количество строчных (маленьких) и прописных (больших)
+букв в введенной строке. Учитывать только английские буквы.
+"""
+
+my_string = 'aAaZ BBbb aa bbb C bbbAqAz bb b b Aaa ББ вв ггОО Q ЖGж'
+print(my_string)
+num_big = 0
+num_small = 0
+for i in range(len(my_string)):
```
Comment (belovmd) [inline src/Homework_02/05_Bukvy.py]: range(len(iterable))

File: src/Homework_02/05_Bukvy.py [hunk 17]
```diff
@@ -0,0 +1,19 @@
...
+    cur_letter = my_string[i]
+    cur_ind = ord(cur_letter)
+    if (65 <= cur_ind <= 90) or (97 <= cur_ind <= 122):
```
Comment (belovmd) [inline src/Homework_02/05_Bukvy.py]: islower() isupper() or 'a'<char<'z'

File: src/Homework_02/Dop_Palindrom.py [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+num = int(input('Введите число\n'))
+num_length = 1
+num_division = num
+while num_division // 10 > 0:
+    num_length += 1
+    num_division //= 10
+print('Длина числа', num_length)
+
+num_x = num
+for i in range(num_length // 2):
```
Comment (belovmd) [inline src/Homework_02/Dop_Palindrom.py:21]: can you just revert number using arithmetic operators?

File: src/Homework_02/Dop_Fibonachi.py [hunk 19]
```diff
@@ -0,0 +1,26 @@
...
+    n_1 = n_2
+    n_2 = n_x + n_2
+    print(n_2, end=' ')
+print('\nЧисло Фибоначчи номер', n, 'равно', n_2)
+
+(num_1, num_2) = (0, 1)
+print('\nВариант 2')
+print(num_1, num_2, end=' ')
+for i in range(2, n):
+    (num_1, num_2) = (num_2, num_1 + num_2)  # более простой вариант
```
Comment (belovmd) [inline src/Homework_02/Dop_Fibonachi.py]: you don't need parentheses

File: src/Homework_02/Dop_Palindrom_Strokoi.py [hunk 20]
```diff
@@ -0,0 +1,10 @@
+# Палиндром с использованием строк
+my_string = 'aaa bbbb aa bbb c bbbaa bb b b aaa'
+print(my_string)
+my_rep_string = my_string.replace(' ', '')
+print(my_rep_string)  # print просто для красоты
```
Comment (belovmd) [inline src/Homework_02/Dop_Palindrom_Strokoi.py]: redundant prints

File: src/Homework_02/Dop_Prostoe_chislo.py [hunk 21]
```diff
@@ -0,0 +1,12 @@
+n = int(input('Введите число\n'))
+if n == 1:
+    print('Число 1 не относится ни к простым, ни к составным числам')
+else:
+    i = 2
+# у составного числа есть делитель, не больше квадратного корня из числа
+    while i**0.5 <= n and n % i != 0:
+        i += 1
+    if i == n:
```
Comment (belovmd) [inline src/Homework_02/Dop_Prostoe_chislo.py]: if is our of while (bad indentation). it wouldn't work

**Review by belovmd — COMMENTED** [review]
1. Overall tasks from external sites pretty easy. You can try to check something more advanced.
2. Please address these issues

File: src/Homework_02/02_Examples(py.checkio.org).py (hunk 1 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(py.checkio.org).py:2]: Done

File: src/Homework_02/02_Examples(py.checkio.org).py (hunk 2 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(py.checkio.org).py]: Done

File: src/Homework_02/02_Examples(www.codewars.com).py (hunk 3 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(www.codewars.com).py]: Fixed

File: src/Homework_02/02_Examples(www.hackerrank.com).py (hunk 4 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(www.hackerrank.com).py]: Task condition

File: src/Homework_02/02_Examples(www.hackerrank.com).py (hunk 5 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(www.hackerrank.com).py]: Task condition

File: src/Homework_02/02_Examples(www.hackerrank.com).py (hunk 6 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(www.hackerrank.com).py]: Done

File: src/Homework_02/02_Examples(www.hackerrank.com).py (hunk 7 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(www.hackerrank.com).py]: Task condition

File: src/Homework_02/02_Examples(www.hackerrank.com).py (hunk 8 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(www.hackerrank.com).py]: Fixed

File: src/Homework_02/02_Examples(www.hackerrank.com).py (hunk 9 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(www.hackerrank.com).py]: Task condition

File: src/Homework_02/02_Examples(www.hackerrank.com).py (hunk 10 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/02_Examples(www.hackerrank.com).py]: Done)

File: src/Homework_02/03_Max_Dlina.py (hunk 11 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/03_Max_Dlina.py]: Fixed

File: src/Homework_02/03_Max_Dlina.py (hunk 12 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/03_Max_Dlina.py]: Done

File: src/Homework_02/03_Max_Dlina.py (hunk 13 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/03_Max_Dlina.py]: Fixed

File: src/Homework_02/04_Povtor.py (hunk 14 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/04_Povtor.py]: Removed range(len(iterable))

File: src/Homework_02/04_Povtor.py (hunk 15 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/04_Povtor.py]: Fixed

File: src/Homework_02/05_Bukvy.py (hunk 16 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/05_Bukvy.py]: Removed range(len(iterable))

File: src/Homework_02/05_Bukvy.py (hunk 17 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/05_Bukvy.py]: Done

File: src/Homework_02/Dop_Palindrom.py (hunk 18 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/Dop_Palindrom.py:21]: Done. It's even easier)

File: src/Homework_02/Dop_Fibonachi.py (hunk 19 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/Dop_Fibonachi.py]: Fixed

File: src/Homework_02/Dop_Palindrom_Strokoi.py (hunk 20 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/Dop_Palindrom_Strokoi.py]: Removed prints

File: src/Homework_02/Dop_Prostoe_chislo.py (hunk 21 above)
Comment (eldar-dniaprouski) [reply src/Homework_02/Dop_Prostoe_chislo.py]: Fixed

File: src/Homework_02/02_Examples(py.checkio.org).py [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+
+"""
+3. Your mission here is to create a function that gets a tuple
+and returns a tuple with 3 elements - the first,
+third and second to the last for the given array.
+"""
+
+
+def easy_unpack(elements: tuple) -> tuple:
+    (a, b, c) = (elements[0], elements[2], elements[-2])
```
Comment (belovmd) [inline src/Homework_02/02_Examples(py.checkio.org).py:52]: redundant parentheses

File: src/Homework_02/03_Max_Dlina.py [hunk 23]
```diff
@@ -0,0 +1,22 @@
+"""
+3. Найти самое длинное слово в введенном предложении.
+Учтите что в предложении есть знаки препинания.
+"""
+
+my_string = 'aaa bbbb! aa, bbb! c bbbaa bb b. b? aaa'
+print(my_string)
+
+punct_string = '!?,.'
```
Comment (belovmd) [inline src/Homework_02/03_Max_Dlina.py:9]: :;()[]'''

File: src/Homework_02/04_Povtor.py [hunk 24]
```diff
@@ -0,0 +1,14 @@
+"""
+4. Вводится строка.
+Требуется удалить из нее повторяющиеся символы и все пробелы.
+Например, если было введено "abc cde def", то должно быть выведено "abcdef".
+"""
+
+# my_string = input('Введите строку символов через пробел\n')
```
Comment (belovmd) [inline src/Homework_02/04_Povtor.py:7]: redundant comments

**Review by belovmd — COMMENTED** [review]
8,5

**Pull request closed without merging**